### PR TITLE
The CopyPaste plugin works properly for merged cells (unmerge them when necessary)

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

So far, the `MergeCells` plugin hasn't been unmerging cells and data population hasn't been intuitive. This PR change that. Handsontable should work the same as Excel from office.com website.

### How has this been tested?

Tested by comparing how HOT and Excel works for different use cases.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. [#1366](https://github.com/handsontable/dev-handsontable/issues/1366)
2. https://github.com/handsontable/handsontable/issues/5092

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [ ] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Review by budnix — CHANGES_REQUESTED** [review]
It mostly looks good. Things to consider:
 * I can not populate the data using an autofill handler after pasting the data. It can be out of the scope of this PR, but maybe not?
Kapture 2023-07-13 at 13 12 57
 * The undo does not restore the merged cells after pasting the data. As mentioned above, this may be out of scope too;
 * In Excel and GS, pasting a single cell to the selection range that contains merge cells makes all cells unmerged. Here the cells are still merged after the action;
Kapture 2023-07-13 at 13 32 22

**Comment by wszymanski** [comment]
> It mostly looks good. Things to consider:

I think that two first problems are out of the scope. The third problem should be fixed. I'm starting doing it, good catch!

**Comment by wszymanski** [comment]
@budnix After deeper consideration I think that both problem should be fixed. It seems that copy paste makes them visible, and we've had no such problems before.

**Comment by wszymanski** [comment]
> It mostly looks good. Things to consider:
> 
> * I can not populate the data using an autofill handler after pasting the data. It can be out of the scope of this PR, but maybe not?
>   Kapture 2023-07-13 at 13 12 57

It seems that the second problem already exist on current HOT (found source of the problem and the simplest case), but I discovered it right after finding a solution. Thus, I've created an [issue](https://github.com/handsontable/handsontable/pull/10457) for it and [PR](https://github.com/handsontable/handsontable/pull/10457) fixing the problem.

**Comment by wszymanski** [comment]
> * In Excel and GS, pasting a single cell to the selection range that contains merge cells makes all cells unmerged. Here the cells are still merged after the action;
>   Kapture 2023-07-13 at 13 32 22

Fixed within https://github.com/handsontable/handsontable/pull/10444/commits/d90453c72f259efbfd74ab7a7295e8642fabb1ca.

**Comment by wszymanski** [comment]
> * The undo does not restore the merged cells after pasting the data. As mentioned above, this may be out of scope too;

I fixed it partially within https://github.com/handsontable/handsontable/pull/10444/commits/69c9fb091d51df4a7a28474c9591e68918138485 and https://github.com/handsontable/handsontable/pull/10444/commits/6ed95269abe06cbdb1b99932dad4aafba7ef1234#diff-b10ff183adacd6a555b05d7337f0f2f432932eb8e3e9f9e37e8fa416335c653aR456-R458. There is yet problem with not batching undo/redo actions. We have to perform two (instead of one) undo actions to see a merged cell (see below).

https://github.com/handsontable/handsontable/assets/141330/9a92f75c-70e7-4ec4-9984-cd5dd24f9740

I think the problem should be done within another issue. We have already the same problem with features which are using at least two HOT's actions.

 For example, `Autofill` plugin enabled by `fillHandle` set to `true` or as an object with `autoInsertRow` property set to `true` creates new rows and fill them with values.

https://github.com/handsontable/handsontable/assets/141330/6224b2a1-b664-439e-a740-0f4f4adca456

A similar usage example is for data pasting which extends the table

https://github.com/handsontable/handsontable/assets/141330/124721b8-6b23-426a-97fd-4fa1e6bbd904

It may be related: https://github.com/handsontable/handsontable/issues/6374. More problems related to `UndoRedo` has been described by @AMBudnik within https://github.com/handsontable/handsontable/issues/7508.

**Comment by wszymanski** [comment]
@budnix I think that review can be done once again.

**Comment by wszymanski** [comment]
I've found a difference.

HOT

https://github.com/handsontable/handsontable/assets/141330/e4e91a23-ef0c-4d45-bfb0-4c5e3fb40f7c

Google Sheets

https://github.com/handsontable/handsontable/assets/141330/40e8f046-0607-4f22-9651-77425304e1b9

Excel online

https://github.com/handsontable/handsontable/assets/141330/25b434b6-a064-4b5b-95da-e680f5e23ae4

File: handsontable/src/plugins/mergeCells/mergeCells.js [hunk 1]
```diff
@@ -152,6 +153,7 @@ export class MergeCells extends BasePlugin {
         return false;
       }
     });
+    this.addHook('beforePaste', (...args) => this.onBeforePaste(...args));
```
Comment (budnix) [inline handsontable/src/plugins/mergeCells/mergeCells.js:156]:
```suggestion
    this.addHook('beforePaste', (...args) => this.#onBeforePaste(...args));
```
It'd be nice to have this defined as private method.

**Review by budnix — APPROVED** [review]

File: handsontable/src/plugins/mergeCells/mergeCells.js (hunk 1 above)
Comment (wszymanski) [reply handsontable/src/plugins/mergeCells/mergeCells.js]: It has been changed within https://github.com/handsontable/handsontable/pull/10444/commits/7155bbfc8c9953edcce21bd725c44f6539fd8de5.

**Pull request merged**